electron/app/components/charts/BarChart.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState, useEffect } from 'react';
+import { Bar } from 'react-chartjs-2';
+import * as sqlite3 from 'sqlite3';
+import log from 'electron-log';
+import interpolateColors from '../../utils/colors';
+
+interface BarChartProps {
+  db: sqlite3.Database;
+  chartQuery: () => Promise<WordCountTypes.Results | TopFriendsTypes.Results>;
+  titleText: string;
+  subLabel: string;
+  xAxisKey: string; // TODO: this should be a type from results
+  stacked?: boolean;
+  colorInterpolationFunc: (t: number) => string;
+}
+
+export default function BarChat(props: BarChartProps) {
+  const {
+    chartQuery,
+    titleText,
+    subLabel,
+    xAxisKey,
+    colorInterpolationFunc,
+  } = props;
+  const [xAxisData, setXAxisData] = useState<string[]>([]);
+  const [count, setCount] = useState<number[]>([]);
+
+  useEffect(() => {
+    async function fetchXAxisData() {
+      try {
+        const dataList = await chartQuery();
+        setXAxisData(
+          (dataList as Array<
+            WordCountTypes.ChartData | TopFriendsTypes.ChartData
+          >).map((obj: { [x: string]: any }) => obj[xAxisKey])
+        );
+        setCount(
+          (dataList as Array<
+            WordCountTypes.ChartData | TopFriendsTypes.ChartData
+          >).map((obj: { [x: string]: any }) => obj.count)
```

Comment:
This is hideous and but I don't know how to get `.map` in Typescript land to not complain about union-ed types